File: API/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Order
    {
        [Key]
        public int OrderId { get; set; }

        public string? BuyerId { get; set; }

        [Required]
        public ShippingAddress? ShippingAddress { get; set; }

         public List<OrderItem>? OrderItems { get; set; }

        //[ForeignKey(nameof(PaymentMethod))]
        public PaymentMethod? PaymentMethod { get; set; }
        public int MethodId { get; set; }

        public long Subtotal { get; set; }

        public long AdditionalExpenses { get; set; }

        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;


         public long GetTotal()
        {
            return Subtotal + AdditionalExpenses;
        }
    }
}

File: client/src/app/models/order.ts
export interface Order {
  orderId: number
  buyerId: string
  shippingAddress: ShippingAddress
  methodId: number
  subtotal: number
  additionalExpenses: number
  orderDate: string
  orderStatus: string
  total: number
  orderItems: OrderItem[]
}

export interface ShippingAddress {
  fullName: string
  address1: string
  address2: string
  address3: string
  city: string
  zip: number
  country: string
}

export interface OrderItem {
  productId: number
  name: string
  imageUrl: string
  price: number
  sizeId: number
  quantity: number
}
